libs/cosmos-sdk/baseapp/baseapp_paralled_test.go

```diff
@@ -0,0 +1,233 @@
+		chain.acc[i] = &types3.EthAccount{
+			BaseAccount: &auth.BaseAccount{
+				Address: env.addr[i],
+				Coins:   sdk.Coins{sdk.NewInt64Coin("okt", 1000000)},
+			},
+			//CodeHash: []byte{1, 2},
+		}
+		genAccs = append(genAccs, chain.acc[i])
+		chain.priv[i] = env.priv[i]
+		chain.addr[i] = env.addr[i]
+		chain.seq[i] = 0
+		chain.num[i] = uint64(i)
+	}
+	chain.chainIdStr = "ethermint-3"
+	chain.chainIdInt = big.NewInt(3)
+
+	chain.app = simapp2.SetupWithGenesisAccounts(genAccs, sdk.NewDecCoins(sdk.NewDecCoinFromDec(sdk.DefaultBondDenom, sdk.NewDecWithPrec(1000000, 0))))
+	//header := abci.Header{Height: app.LastBlockHeight() + 1, ChainID: chainIdStr}
+
+	chain.app.BaseApp.Commit(abci.RequestCommit{})
+	return chain
+}
+
+func createEthTx(t *testing.T, chain *Chain, i int) []byte {
+	amount, gasPrice, gasLimit := int64(1024), int64(2048), uint64(100000)
+	addrTo := ethcmn.BytesToAddress(chain.priv[i+1].PubKey().Address().Bytes())
+	msg := types.NewMsgEthereumTx(chain.seq[i], &addrTo, big.NewInt(amount), gasLimit, big.NewInt(gasPrice), []byte("test"))
+	chain.seq[i]++
+	err := msg.Sign(chain.chainIdInt, chain.priv[i].ToECDSA())
+	require.NoError(t, err)
+	rawtx, err := rlp.EncodeToBytes(&msg)
+	require.NoError(t, err)
+
+	return rawtx
+}
+
+func createCosmosTx(t *testing.T, chain *Chain, i int) []byte {
+	msg := types4.NewMsgTokenSend(chain.addr[i], chain.addr[i+1], sdk.Coins{sdk.NewInt64Coin("okt", 10)})
+
+	tx := helpers.GenTx(
+		[]sdk.Msg{msg},
+		sdk.Coins{sdk.NewInt64Coin(sdk.DefaultBondDenom, 1)},
+		helpers.DefaultGenTxGas,
+		chain.chainIdStr,
+		[]uint64{chain.num[i]},
+		[]uint64{chain.seq[i]},
+		chain.priv[i],
+	)
+	chain.seq[i]++
+
+	txBytes, err := chain.app.Codec().MarshalBinaryLengthPrefixed(tx)
+	require.Nil(t, err)
+	return txBytes
+}
+
+func runtxs(chain *Chain, rawTxs [][]byte, isParalle bool) []*abci.ResponseDeliverTx {
+	header := abci.Header{Height: chain.app.LastBlockHeight() + 1, ChainID: chain.chainIdStr}
+	chain.app.BaseApp.BeginBlock(abci.RequestBeginBlock{Header: header})
+	ret := []*abci.ResponseDeliverTx{}
+	if isParalle {
+		ret = chain.app.BaseApp.ParallelTxs(rawTxs, false)
+	} else {
+		for _, tx := range rawTxs {
+			r := chain.app.BaseApp.DeliverTx(abci.RequestDeliverTx{Tx: tx})
+			ret = append(ret, &r)
+		}
+	}
+	chain.app.BaseApp.EndBlock(abci.RequestEndBlock{})
+	chain.app.BaseApp.Commit(abci.RequestCommit{})
+
+	return ret
+}
+
+func TestParalledTxs(t *testing.T) {
+	env := new(Env)
+	accountNum := 10
+	env.priv = make([]ethsecp256k1.PrivKey, 10)
+	env.addr = make([]sdk.AccAddress, 10)
+	for i := 0; i < accountNum; i++ {
+		priv, _ := ethsecp256k1.GenerateKey()
+		addr := sdk.AccAddress(priv.PubKey().Address())
+		env.priv[i] = priv
+		env.addr[i] = addr
+	}
+
+	chainA, chainB := NewChain(env), NewChain(env)
+
+	testCases := []struct {
+		name     string
+		malleate func(t *testing.T, chain *Chain, isParallel bool) ([]byte, []byte)
+	}{
+		{
+			"five txs one group:a->b b->c c->d d->e e->f",
```

Comment:
maybe add some scenarios like tx failed

Reply:
0fd6a96a9a27899dd2a143afb26f140b6b7d022e has add two case with conflicted in contract